### PR TITLE
Clean up hacky routes

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
- Provider code as URL

### Changes proposed in this pull request
- Add `friendly_id` to use the `provider_code` as the slug inside of the `id`

### CODE REVIEW

File: app/controllers/api/v2/courses_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module API
   module V2
     class CoursesController < ApplicationController
       def index
-        provider = Provider.find_by!(provider_code: params[:provider_code])
+        provider = Provider.friendly.find(params[:provider_id])
```
Comment (defong) [inline app/controllers/api/v2/courses_controller.rb]: `params[:provider_id]` so is that the `friendly_id` aka `provider_code`?

Comment (dankmitchell) [reply app/controllers/api/v2/courses_controller.rb]: Yes. We define what the slug/url in the model so the controllers and routes are standard rails with the except of the `friendly` helper in the controller to hook it all together.

**Comment by dankmitchell** [comment]
@timabell I dont know where those commits came from but ive cleaned it up

**Comment by defong** [comment]
@dankmitchell long live pr?

**Comment by dankmitchell** [comment]
Closing as I know @misaka isn’t keen

**Pull request merged**